src/api/app/controllers/webui/cloud/ec2/configurations_controller.rb

```diff
@@ -0,0 +1,36 @@
+module Webui
+  module Cloud
+    module Ec2
+      class ConfigurationsController < WebuiController
+        before_action :require_login, :set_user
+        before_action -> { feature_active?(:cloud_upload) }
+
+        def show
+          @crumb_list = ['EC2 Configuration']
+          @ec2_configuration = @user.ec2_configuration || @user.create_ec2_configuration
+          @aws_account_id = CONFIG['aws_account_id']
+        end
+
+        def update
+          if @user.ec2_configuration.update(permitted_params)
+            flash[:success] = 'Successfully updated your EC2 configuration.'
+            redirect_to cloud_upload_index_path
+          else
+            flash[:error] = "Failed to updated your EC2 configuration: #{@user.ec2_configuration.errors.full_messages.to_sentence}."
+            redirect_back(fallback_location: root_path)
+          end
+        end
+
+        private
+
+        def permitted_params
+          params.require(:ec2_configuration).permit(:id, :arn)
+        end
+
+        def set_user
+          @user = User.current
+        end
```

Comment:
This seems to be a generic helper. I would move it to the webui controller.

Though I don't understand why we don't directly use User.current.

Reply:
Removed....